src/Integration.Vsix.UnitTests/CFamily/CFamilyHelperTests.cs

```diff
@@ -733,4 +799,46 @@ string IVCRulePropertyStorage.GetEvaluatedPropertyValue(string propertyName)
         }
     }
 
+    public class VCProjectMock
+    {
+        public VCConfigurationMock Configurations { get; set; } = new VCConfigurationMock();
+        public VCConfigurationMock FileConfigurations { get; set; } = new VCConfigurationMock();
+        public VCConfigurationMock ActiveConfiguration { get; set; } = new VCConfigurationMock();
+        public VCProjectMock Object => this;
+    }
+
+    public class VCConfigurationMock : IVCRulePropertyStorage
+    {
+        public VCConfigurationMock Rules => this;
+        public VCConfigurationMock Tool => this;
+        public string PlatformToolset => "v140_xp";
+        public string PrecompiledHeader => "NotUsing";
+        public string CompileAs => "CompileAsCpp";
+        public string CompileAsManaged => "false";
+        public string BasicRuntimeChecks => "UninitializedLocalUsageCheck";
+        public string ExceptionHandling => "Sync";
+        public string EnableEnhancedInstructionSet => "";
+        public string RuntimeLibrary => "";
+        public string LanguageStandard => "";
+
+        string IVCRulePropertyStorage.GetEvaluatedPropertyValue(string propertyName)
+        {
+            return GetEvaluatedPropertyValue(propertyName);
+        }
+
+        public string GetEvaluatedPropertyValue(string propertyName)
+        {
+            var propertyInfo = GetType().GetProperty(propertyName);
+            if (propertyInfo != null)
+            {
+                return (string)propertyInfo.GetValue(this);
+            }
+            return "dummy property value";
+        }
+
+        public dynamic Item(string name)
```

Comment:
As discussed offline, here a couple of things to check/do:
* move the classes out of this ugly hack namespace if they don't need to be there
* check whether `dynamic Item` is required, and if it needs to be dynamic
* replace the properties that don't need to be accessed directly as properties with a dictionary, and change `GetEvaluatedPropertyValue` to use the dictionary rather than reflection.